core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -140,15 +150,50 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
 
         createdContainer match {
           case Some(((actor, data), containerState)) =>
+            //increment active count before storing in pool map
+            val (newData, container) = data match {
+              case p: PreWarmedData => //convert PreWarmedData -> WarmingData for concurrent cases
```

Comment:
The state semantics are bit confusing here. `NoData` transitions to `WarmingColdData` for concurrent case but remains `NoData` for non concurrent case. Can we have uniform transition here and adapt the logic at other places. 

* {NoData, MemoryData} ->  WarmingColdData -> WarmedData
* PreWarmedData -> WarmingData -> WarmedData

Also may be we make this logic somewhat polymorphic i.e. have the data instance "transition" to new data itself and move this logic there